py_async_grpc_prometheus/prometheus_async_client_interceptor.py

```diff
@@ -8,185 +8,382 @@
+            client_call_details,
+            grpc_utils.wrap_iterator_inc_counter(
+                request_iterator,
+                iterator_sent_metric,
+                grpc_type,
+                grpc_service_name,
+                grpc_method_name,
+            ),
+        )
+
+        if self._enable_client_stream_send_time_histogram and not self._legacy:
+            self._metrics["grpc_client_stream_send_histogram"].labels(
+                grpc_type=grpc_type,
+                grpc_service=grpc_service_name,
+                grpc_method=grpc_method_name,
+            ).observe(max(default_timer() - start, 0))
+
+        iterator_received_metric = self._metrics["grpc_client_stream_msg_received"]
+
+        response_iterator = grpc_utils.wrap_iterator_inc_counter(
+            response_iterator,
+            iterator_received_metric,
             grpc_type,
             grpc_service_name,
-            grpc_method_name))
-
-    if self._enable_client_stream_send_time_histogram and not self._legacy:
-      self._metrics["grpc_client_stream_send_histogram"].labels(
-          grpc_type=grpc_type,
-          grpc_service=grpc_service_name,
-          grpc_method=grpc_method_name).observe(max(default_timer() - start, 0))
-
-    iterator_received_metric = self._metrics["grpc_client_stream_msg_received"]
-
-    response_iterator = grpc_utils.wrap_iterator_inc_counter(
-        response_iterator,
-        iterator_received_metric,
-        grpc_type,
-        grpc_service_name,
-        grpc_method_name)
-
-    if self._enable_client_stream_receive_time_histogram and not self._legacy:
-      self._metrics["grpc_client_stream_recv_histogram"].labels(
-          grpc_type=grpc_type,
-          grpc_service=grpc_service_name,
-          grpc_method=grpc_method_name).observe(max(default_timer() - start, 0))
-
-    return response_iterator
+            grpc_method_name,
+        )
+
+        if self._enable_client_stream_receive_time_histogram and not self._legacy:
+            self._metrics["grpc_client_stream_recv_histogram"].labels(
+                grpc_type=grpc_type,
+                grpc_service=grpc_service_name,
+                grpc_method=grpc_method_name,
+            ).observe(max(default_timer() - start, 0))
+
+        return response_iterator
+
+
+class PromAsyncClientInterceptor(_PromAsyncUnaryUnaryClientInterceptor,
+                            _PromAsyncUnaryStreamClientInterceptor,
+                            _PromAsyncStreamUnaryClientInterceptor,
+                            _PromAsyncStreamStreamClientInterceptor):
+    """
+    Intercept gRPC client requests.
+    """
+    def __init__(
+        self,
+        enable_client_handling_time_histogram=False,
+        enable_client_stream_receive_time_histogram=False,
+        enable_client_stream_send_time_histogram=False,
+        legacy=False,
+        registry=REGISTRY,
+    ):
+        metrics = init_metrics(REGISTRY)
+        _PromAsyncUnaryUnaryClientInterceptor.__init__(
```

Comment:
Why not do `_PromAsyncUnaryUnaryClientInterceptor()` etc in this case?

Reply:
Because this is instantiating each of the parent classes. It would be like calling `super().__init__(...)` if there was only one parent class. And super works differently in multiple inheritance.